pkg/types/gcp/platform.go

```diff
@@ -29,4 +29,9 @@ type Platform struct {
 	// The value should be the name of the subnet.
 	// +optional
 	ComputeSubnet string `json:"computeSubnet,omitempty"`
+
+	// Licenses is a list of licenses to apply to the compute images
+	// The value should a list of strings representing the license keys
```

Comment:
1. add the requirement that these are URLs (https) only
2. add documentation that this field cannot be used to any kind of OS image override
^^ also update the docs to say that.. we don't want to enumerate the it's only the env, but generically all workflows like in future the clusterOSImage fields.
3. also add documentation that when set, this will cause the installer to copy the image into user's project.

Reply:
looking at the docs https://cloud.google.com/compute/docs/reference/rest/v1/images/insert

```
licenses[] | string                     Any applicable license URI.                     
Authorization requires the following IAM permission on the specified resource licenses:                                            compute.licenseCodes.use
```

it seems like the user creating this needs to have certain permissions to use it. we should add that to docs/


there is also an API to get the license.. https://cloud.google.com/compute/docs/reference/rest/v1/licenses/get
we should add a validation to `pkg/asset/installconfig/gcp/validations.go` such that it can get that the user can GET the license and the users are not setting invalid licenses..

Reply:
OK

Reply:
Giving it a try, I think it might not be possible to validate the nested license through the compute.license API. Such API seems to be intended to inspect project-specific resources:
`GET https://compute.googleapis.com/compute/v1/projects/{project}/global/licenses/{resourceId}`

However, the nested license seems to live in a special internal "project" called "vm-options".
> https://www.googleapis.com/compute/v1/projects/vm-options/global/licenses/enable-vmx

Trying to access that resource from the API yields a permission error. And trying to grant permissions over this "vm-options" project also fails.
I have talked with GCP's tech support and they have confirmed this.

Should we still try to keep the option generic (string-based license URLs) or should we add a "enable-nested" knob?

Reply:
@abhinavdahiya ^^

Reply:
it looks like we can't validate if license is valid, so let's drop that specific validation.